lib/json_api_controller/json_schema_validator.rb

```diff
@@ -7,7 +7,8 @@ module JsonSchemaValidator
 
     included do
       @action_params = Hash.new
-      (@actions & %i(update create)).each do |action|
+      schema_actions = %i(update create) | Array.wrap(@extra_schema_actions)
+      (@actions | schema_actions).each do |action|
```

Comment:
because there is no associated wired up action from the `resource_actions` DSL, you'll need to override the preferences controller shared module resource_actions to include your update_settings method too. 

And yes this should be the intersection otherwise it'll run these for GET req's etc and it only need to validate the schema on POST / PUT methods.